staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -304,6 +304,10 @@ func (rules *ClientConfigLoadingRules) Migrate() error {
 
 // GetLoadingPrecedence implements ConfigAccess
 func (rules *ClientConfigLoadingRules) GetLoadingPrecedence() []string {
+	if len(rules.ExplicitPath) > 0 {
+		return []string{rules.ExplicitPath}
```

Comment:
seems like when you set this field, perhaps you should reset rules.Precendence?  At least from genericclioptions?

Reply:
I tried, but it looks like this is breaking all of our test-cmd, so I don't think it's the right fix. With proper LoadingPrecedence we should be good, though.

Reply:
it is also special cased above in `Load` so I guess I can live with it.  It doesn't seem awesome though.

Reply:
I'm planning to squashing all `GetLoadingPrecedence` methods into single execution path, to avoid duplicated code and problems like the one this PR is fixing. But I wanted to make sure that approach isn't breaking anything.